pkg/governance-scripts/contracts/20220322-veBAL-activation/veBALDeploymentCoordinator.sol

```diff
@@ -187,19 +221,68 @@ contract veBALDeploymentCoordinator is ReentrancyGuard {
             authorizer.revokeRole(authorizerAdaptor.getActionId(IGaugeController.add_type.selector), address(this));
         }
 
-        // Step 4: create gauges for the single-gauge gauge types
+        // Step 5: setup the GaugeAdder contract to be in charge of adding gauges to the Gauge Controller.
         //
-        // The LM committee and veBAL gauge types will have a single gauge of the single-recipient gauge kind
+        // The GaugeAdder contract performs checks on addresses being added to the Gauge Controller to ensure
+        // that they have been deployed by a factory contract which has been associated with the gauge type
+        // to which the proposed gauge is being added. This is intended to prevent common mistakes when adding gauges.
 
-        // grant self power to add gauges
-        // deploy lm, vebal, merkle gauges. add gauges to gauge controller
+        authorizer.grantRole(authorizerAdaptor.getActionId(IGaugeController.add_gauge.selector), address(_gaugeAdder));
 
-        // grant gauge adder authority over controller
-        // grant self power over guage adder
-        // add l1 and l2 factories to gauge adder
-        // deploy l1 and l2 gauges, add to controller via gauge adder
+        // Step 6: create gauges for a preselected list of pools on Ethereum.
+
+        // Allowlist the provided LiquidityGaugeFactory on the GaugeAdder
+        // so its gauges may be added to the "Ethereum" gauge type.
+        {
+            authorizer.grantRole(_gaugeAdder.getActionId(IGaugeAdder.addGaugeFactory.selector), address(this));
+
+            _gaugeAdder.addGaugeFactory(_ethereumGaugeFactory, IGaugeAdder.GaugeType.Ethereum);
+
+            authorizer.revokeRole(_gaugeAdder.getActionId(IGaugeAdder.addGaugeFactory.selector), address(this));
+        }
+
+        // Deploy initial gauges and add them to the Gauge Controller
+        {
+            authorizer.grantRole(_gaugeAdder.getActionId(IGaugeAdder.addEthereumGauge.selector), address(this));
+
+            uint256 poolsLength = _initialPools.length;
+            for (uint256 i = 0; i < poolsLength; i++) {
+                ILiquidityGauge gauge = _ethereumGaugeFactory.deploy(_initialPools[i]);
+                _gaugeAdder.addEthereumGauge(address(gauge));
+            }
+
+            authorizer.revokeRole(_gaugeAdder.getActionId(IGaugeAdder.addEthereumGauge.selector), address(this));
```

Comment:
Is there any account that we want to keep this permission? Or are we deploying all sensible gauges already? What if new pools are created after deployment?

Reply:
Added an issue for this. https://github.com/balancer-labs/balancer-v2-monorepo/issues/1190